R/format_ci.R

```diff
@@ -0,0 +1,137 @@
+#' Format an Estimate and Confidence Interval as Text
+#'
+#' The function rounds the estimate, lower and upper confidence interval to the
+#' same magnitude.
+#' The magnitude shows the width of the confidence interval with two significant
+#' digits.
+#' @param estimate The estimate in the `link` scale.
+#' @param se The standard error in the `link` scale.
+#' If missing, you must `lcl` and `ucl`.
+#' @param lcl The lower confidence limit.
+#' Ignored when `se` is given.
+#' @param ucl The upper confidence limit.
+#' Ignored when `se` is given.
+#' @param interval The coverage of the confidence interval.
+#' Only used when `se` is given.
+#' Defaults to `0.95` (`95%`).
+#' @param link The transformation of `estimate`, `se`, `lcl` and `ucl`.
+#' The appropriate backtransformation is applied before formatting.
+#' @param max_digit The maximum number of significant digits to display.
+#' Defaults to `4`.
+#' @param percent Display the interval as a percentage
+#' (= muliply by 100 and append `%`).
+#' Defaults to `FALSE`.
+#' @param sign Always add the sign to the text. (e.g. `+1` instead of `1`).
+#' Defaults to `FALSE`.
+#' @export
+#' @importFrom assertthat assert_that is.flag is.number noNA
+#' @importFrom stats plogis qnorm
+#' @family display
+#' @examples
+#' format_ci(0.512345, 1)
+#' format_ci(0.512345, 1, interval = 0.9)
+#' format_ci(0.512345, 1, link = "log")
+#' format_ci(0.512345, 1, link = "logit")
+#' format_ci(0.512345, 10)
+#' format_ci(0.512345, 0.1)
+#' format_ci(0.512345, 0.01)
+#' format_ci(0.512345, 0.001)
+#' format_ci(0.512345, 0.0001)
+#' format_ci(0.512345, 0.00001)
+#' format_ci(0.512345, 0.00001, max_digit = 10)
+#' format_ci(0.512345, 0.5)
+#' format_ci(-0.1, lcl = -0.1999, ucl = 0.1234)
+#' format_ci(-0.1, lcl = -0.1999, ucl = 0.1234, percent = TRUE)
+#' format_ci(-0.1, lcl = -0.1999, ucl = 0.1234, sign = TRUE)
+#' format_ci(-0.1, lcl = -0.1999, ucl = 0.1234, percent = TRUE, sign = TRUE)
+#' format_ci(0.512345e-6, 1e-6)
+#' format_ci(0.512345e-7, 1e-7)
+#' format_ci(0.512345e-7, 1e-8)
+#' format_ci(0.512345e-7, 1e-9)
+format_ci <- function(
+  estimate, se, lcl, ucl, interval = 0.95, link = c("identity", "log", "logit"),
+  max_digit = 4, percent = FALSE, sign = FALSE
+) {
+  link <- match.arg(link)
+  assert_that(
+    is.numeric(estimate),
+    noNA(estimate),
+    is.number(max_digit),
+    is.flag(percent),
+    noNA(percent),
+    is.flag(sign),
+    noNA(sign)
+  )
+  if (missing(se)) {
+    assert_that(
+      is.numeric(lcl),
+      is.numeric(ucl),
+      noNA(lcl),
+      noNA(ucl),
+      all(lcl <= estimate),
+      all(estimate <= ucl)
+    )
+  } else {
+    assert_that(
+      missing(lcl),
+      missing(ucl),
+      is.numeric(se),
+      noNA(se),
+      all(0 <= se),
+      is.number(interval),
+      0 < interval,
+      interval < 1
+    )
+    lcl <- qnorm((1 - interval) / 2, mean = estimate, sd = se)
+    ucl <- qnorm(0.5 + interval / 2, mean = estimate, sd = se)
```

Comment:
The same but a little bit clearer wrt the previous line:

```suggestion
    ucl <- qnorm((1 + interval) / 2, mean = estimate, sd = se)
```